bin/tx-prover/src/proxy/mod.rs

```diff
@@ -154,77 +142,73 @@ impl ProxyHttp for LoadBalancer {
         let client_addr = session.client_addr();
         let user_id = client_addr.map(|addr| addr.to_string());
 
-        // Request ID is a random number
-        let request_id = rand::random::<u64>().to_string();
-        session.req_header_mut().insert_header("X-Request-ID", request_id)?;
-
         // Retrieve the current window requests
         let curr_window_requests = RATE_LIMITER.observe(&user_id, 1);
 
+        // Rate limit the request
         if curr_window_requests > self.max_req_per_sec {
             return Self::create_too_many_requests_response(session, self.max_req_per_sec).await;
         };
+
+        // Check if the queue is full
+        {
+            let queue = QUEUE.read().await;
+            if queue.len() >= self.max_queue_items {
+                return Self::create_queue_full_response(session).await;
```

Comment:
nit: Since these functions don't need references to `self`, might be better to move them outside of `LoadBalancer` and make them helper functions